arches_project/core/arches/connection.py

```
@@ -145,9 +153,11 @@ def get_graphs(self, arches_graphs_list, login_updates, percent_progress):
                                 "multiple_geometry_nodes": multiple,
                             }
                         )
-                    percent_progress.emit(False, x + 1, len(graphids))
-        except:
-            pass
+                    if percent_progress:
+                        percent_progress.emit(False, x + 1, len(graphids))
+
+        except Exception as e:
+            print("Exception", e)
```

Arguably I should remove this, if we're avoiding catch all exception printing?

---

Yeah we should replace this Exception with the actual type of error that will be thrown

---

This spurred quite a lot more thinking about the ‘bad path’, specifically what happens when an Arches connection stops after you’ve already logged in, and you try and refresh the connection. 

I’ve now caught any 4XX or 5XX errors in `get_graph` using `raise_for_status.` We’d want to handle this quite differently in the unlikely event it occurred as part `ConnectionProcess`, so I’ve raised these exceptions up to where they're called. 

Here, I’ve modified `update_refresh_confirm_label`to say the refresh failed if the connection argument is `False`. I also thought it should be given the same prominence as a failed initial connection, so I’ve shown a message in the banner on the QGIS interface. 

In addition, I thought we probably don’t want people to continue making resource requests if the `connection_refresh` has failed, and the edit tab shouldn’t still say ‘Arches is connected’. But nor do we want to log them out. 

`connection_reset`, with `hard_reset=False` wasn’t being used anywhere, so for now I’ve adopted this with some modifications. `arches_api` is now only reset when `hard_reset=True`. This means a ‘soft reset’ disables the resource functionality, but leaves the user logged in, and doesn’t wipe all the connection data. If the connection is re-established and the connection is refreshed again, the user can carry on as before.